pkg/config/config.go

```diff
@@ -37,9 +37,53 @@ type ConfigurationBuilder struct {
 // Unmarshal loads configuration into the provided structure from environment variables.
 // Use the "env" tag on cfgStruct fields to indicate the corresponding environment variable to load from.
 func (config *ConfigurationBuilder) Unmarshal(cfgStruct interface{}) error {
+	// Unmarshal env vars into the struct
 	config.parsers = config.createCustomParsers()
 	err := env.ParseWithFuncs(cfgStruct, config.parsers)
-	return err
+	if err != nil {
+		return err
+	}
+
+	// Unmarshal services into the struct
+	val := reflect.ValueOf(cfgStruct)
+	// Reflection will panic on nil cfgStruct, so better to catch it here
+	if val.IsNil() {
+		return errors.New("Unable to unmarshal config: the provided struct is nil")
+	}
+	if config.values != nil {
+		elem := val.Elem()
+		// Look at all the fields in our target struct
+		for fi := 0; fi < elem.NumField(); fi++ {
+			field := elem.Field(fi)
+			fieldType := field.Type()
+
+			// See if we have a matching service type
+			for ti, t := range config.values.types {
```

Comment:
Some of this might be duplicating code in `GetService`, but I couldn't figure out how to use `GetService` here, and gave up after spending too much time on it.
If anyone else can get it to work, I'm open to it.